processing/src/main/java/io/druid/segment/LongMetricColumnSerializer.java

```diff
@@ -51,8 +54,8 @@ public LongMetricColumnSerializer(
   @Override
   public void open() throws IOException
   {
-    writer = CompressedLongsSupplierSerializer.create(
-        ioPeon, String.format("%s_little", metricName), IndexIO.BYTE_ORDER,
+    writer = CompressionFactory.getLongSerializer(
+        ioPeon, String.format("%s_little", metricName), IndexIO.BYTE_ORDER, CompressionFactory.DEFAULT_LONG_ENCODING,
```

Comment:
previously segments created by using IndexMerger then V8toV9 conversion didn't use compression strategy specified by the indexSpec, so I also used the default encoding. Changed it in the newer commit to use compression and encoding from indexSpec instead